src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -816,21 +816,22 @@ public abstract partial class JsonSerializerContext
         public abstract System.Text.Json.Serialization.Metadata.JsonTypeInfo? GetTypeInfo(System.Type type);
     }
     [System.AttributeUsageAttribute(System.AttributeTargets.Class, AllowMultiple=false)]
-    public partial class JsonSerializerOptionsAttribute : System.Text.Json.Serialization.JsonAttribute
+    public partial class JsonSourceGenerationOptionsAttribute : System.Text.Json.Serialization.JsonAttribute
     {
-        public JsonSerializerOptionsAttribute() { }
+        public JsonSourceGenerationOptionsAttribute() { }
         public System.Text.Json.Serialization.JsonIgnoreCondition DefaultIgnoreCondition { get { throw null; } set { } }
         public bool IgnoreReadOnlyFields { get { throw null; } set { } }
         public bool IgnoreReadOnlyProperties { get { throw null; } set { } }
         public bool IgnoreRuntimeCustomConverters { get { throw null; } set { } }
         public bool IncludeFields { get { throw null; } set { } }
-        public System.Text.Json.Serialization.JsonKnownNamingPolicy NamingPolicy { get { throw null; } set { } }
+        public System.Text.Json.Serialization.JsonKnownNamingPolicy PropertyNamingPolicy { get { throw null; } set { } }
         public bool WriteIndented { get { throw null; } set { } }
+        public JsonSourceGenerationMode GenerationMode { get { throw null; } set { } }
     }
     [System.FlagsAttribute]
     public enum JsonSourceGenerationMode
     {
-        MetadataAndSerialization = 0,
+        Unspecified = 0,
```

Comment:
Totally my personal opinion, but I think I like `Default` better than `Unspecified`.

Reply:
Unspecified here means that no generation setting was specified (as opposed to metadata + serialization like done before). My thought is that the default value of `JsonSerializableAttribute.GenerationMode` (type-specific) is `Unspecified`, meaning that the context-wide `JsonSourceGenerationOptions.GenerationMode` (with the _property's_ default value being metadata + serialization) would win.

Reply:
Why can't `Default` at the type-specific level mean "use the context-wide setting"?

Reply:
That should be fine. A separate question - if we add a `Deserialization` mode for fast-path with the reader in the future, should the `Default` behavior be updated to include deserialization logic, or would that be an unacceptable breaking change?

Reply:
It wouldn't be a breaking change to add more code by default.

But yes, updating the behavior of "Default" would be just as acceptable as updating the behavior of "Unspecified", as long as it isn't breaking.

Reply:
I've updated the `Unspecifed` -> `Default` naming and behavior, as suggested. PTAL @ https://github.com/dotnet/runtime/pull/54527/commits/93a1da7c391e178c0c1906dd724c0babf717a00e.